integreat_cms/static/src/js/pages/page-api-token.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Generate and (un)set the API token.
+ */
+
+import { copyToClipboard } from "../copy-clipboard";
+
+// Set event handler
+window.addEventListener("load", () => {
+    document.getElementById("checkbox_api_token")?.addEventListener("change", setApiToken);
+    let checkbox = document.getElementById("checkbox_api_token") as HTMLInputElement;
+    document.getElementById("copy-api-token")?.addEventListener("click", copyApiToken);
+});
+
+/**
+ * Generate 24 characters API token and insert it into the text field
+ *
+ * @param {target} eventTarget - The enable api token checkbox
+ */
+function setApiToken({ target }: Event) {
+    let checkBox = target as HTMLInputElement;
+    let text = document.getElementById("id_api_token") as HTMLInputElement;
+    let apiTokenContainer = document.getElementById("api-token-container") as HTMLInputElement;
+    if (checkBox.checked){
+      text.value = window.crypto.randomUUID();
+      apiTokenContainer.classList.remove("hidden");
+    } else {
+      text.value = "";
+      apiTokenContainer.classList.add("hidden");
+    }
+}
```

Comment:
```suggestion
}

/**
 * Copy API token to clipboard
 *
 * @param event Event - The click event
 */
function copyApiToken(event: Event) {
    event.preventDefault();
    let text = document.getElementById("id_api_token") as HTMLInputElement;
    let copyButton = document.getElementById("copy-api-token");
    let successBox = document.getElementById("copy-api-token-success");
    // Copy value to clipboard
    copyToClipboard(text.value);
    // Indicate success by changing button
    copyButton.classList.add("hidden");
    successBox.classList.remove("hidden");
    // Revert button after 3 seconds
    setTimeout(function() {
        successBox.classList.add("hidden");
        copyButton.classList.remove("hidden");
    }, 3 * 1000);
}

```
Oh, I forgot to suggest my `copyApiToken()` function.

Reply:
Thanks, I'll copy this.